solr/core/src/java/org/apache/solr/core/PluginBag.java

```diff
@@ -140,11 +139,10 @@ public static void initInstance(Object inst, PluginInfo info) {
       log.debug("{} : '{}' created with startup=lazy ", meta.getCleanTag(), info.name);
       return new LazyPluginHolder<T>(meta, info, core, core.getResourceLoader(), false);
     } else {
-      if (info.pkgName != null) {
-        PackagePluginHolder<T> holder = new PackagePluginHolder<>(info, core, meta);
-        return holder;
+      if (core.getResourceLoader().getPackage(info.pkgName) != null) {
```

Comment:
that method SolrResourceLoader.getPackage will fail fast it can't load it for whatever reason.  See `handleUnresolvedPackage` which it calls.

Reply:
The idea of `SRL` trying to resolve the correct package is wrong. Let's not do this. SRL has no idea what version of a package should be used to load a plugin. Only a `SolrCore` does

Reply:
> The idea of SRL trying to resolve the correct package is wrong.

You seem pretty adamant about this; I suggest saying you don't like that approach instead of it being "wrong".  In this PR, SRL resolving the package is a central theme and so if we don't agree on wether SRL should do this then this PR has no future (ah well).  Maybe we will return to this idea if there are holes in other approaches.  

I view the lack of correct version resolution in the PR (still only one commit as I write this) as merely a necessary TODO, and I don't forsee it'd be hard.  If you would like me to explore adding this next, let me know.